tcn_hpl/data/vectorize/locs_and_confs.py

```diff
@@ -0,0 +1,141 @@
+        top_k: The number of top per-class examples to use in vector
+            construction.
+    """
+
+    def __init__(
+        self,
+        top_k: int = 1,
+        num_classes: int = 7,
+        use_joint_confs: bool = True,
+        use_pixel_norm: bool = True,
+        use_hand_obj_offsets: bool = False,
+        background_idx: int = 0
+    ):
+        super().__init__()
+
+        self._top_k = top_k
+        self._num_classes = num_classes
+        self._use_joint_confs = use_joint_confs
+        self._use_pixel_norm = use_pixel_norm
+        self._use_hand_obj_offsets = use_hand_obj_offsets
+        self._background_idx = background_idx
+    
+    # Get the top "k" object indexes for each object
+    @staticmethod
+    def get_top_k_indexes_of_one_obj_type(f_dets, k, label_ind):
+        """
+        Find all instances of a label index in object detections.
+        Then sort them and return the top K.
+        Inputs:
+        - object_dets:
+        """
+        labels = f_dets.labels
+        scores = f_dets.scores
+        # Get all labels of an obj type
+        filtered_idxs = [i for i, e in enumerate(labels) if e == label_ind]
+        if not filtered_idxs:
+            return None
+        filtered_scores = [scores[i] for i in filtered_idxs]
+        # Sort labels by score values.
+        sorted_inds = [i[1] for i in sorted(zip(filtered_scores, filtered_idxs))]
+        return sorted_inds[:k]
+    @staticmethod
+    def append_vector(frame_feat, i, number):
+        frame_feat[i] = number
+        return frame_feat, i + 1
+
+
+    def vectorize(self, data: FrameData) -> npt.NDArray[np.float32]:
+
+        #########################
+        # Feature vector
+        #########################
+        # Length: pose confs * 22, pose X's * 22, pose Y's * 22,
+        #         obj confs * num_objects(7 for M2), 
+        #         obj X * num_objects(7 for M2), 
+        #         obj Y * num_objects(7 for M2)
+        #         obj W * num_objects(7 for M2)
+        #         obj H * num_objects(7 for M2)
+        #         casualty conf * 1
+        vector_len = 102
+        frame_feat = np.zeros(vector_len)
+        vector_ind = 0
+        if self._use_pixel_norm:
+            W = data.size[0]
+            H = data.size[1]
+        else:
+            W = 1
+            H = 1
+        f_dets = data.object_detections
+
+        # Loop through all classes: populate obj conf, obj X, obj Y.
+        # Assumption: class labels are [0, 1, 2,... num_classes-1].
+        for obj_ind in range(0,self._num_classes):
+            top_k_idxs = self.get_top_k_indexes_of_one_obj_type(f_dets, self._top_k, obj_ind)
+            if top_k_idxs: # This is None if there were no detections to sort for this class
+                for idx in top_k_idxs:
+                    # Conf
+                    frame_feat, vector_ind = self.append_vector(frame_feat, vector_ind, f_dets.scores[idx])
```

Comment:
If we're more fixing vector construction, would we be able to get away from this incrementing index counter? I.e. could we come into this loop with some offset (if any in this case) and directly assign values to computed indices?
E.g.
```python
offset = 0
for ...:
    obj_offset = offset + (self._num_classes * self._top_k * obj_ind)
    top_k_idxs = ...
    if top_k_idxs:
        for i, det_idx in enumerate(top_k_idxs):
            topk_offset = self._top_k * i
            frame_feat[obj_offset + topk_offset + 0] = f_dets.scores[det_idx]
            frame_feat[obj_offset + topk_offset + 1] = f_dets.boxes[det_idx][0] / W
            ...
```

Reply:
Just for now add a TODO that we want to move to direct assignment.

Reply:
Done.